apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -1027,6 +1027,136 @@ defmodule Astarte.RealmManagement.Queries do
     end
   end
 
+  def get_interfaces_details(client) do
+    interfaces_details_query = """
+    SELECT * FROM interfaces
+    """
+
+    query =
+      DatabaseQuery.new()
+      |> DatabaseQuery.statement(interfaces_details_query)
+      |> DatabaseQuery.consistency(:quorum)
+
+    with {:ok, interfaces_result} <- DatabaseQuery.call(client, query) do
+      interfaces =
+        Enum.map(interfaces_result, fn interface_row ->
+          %{
+            name: name,
+            major_version: major_version,
+            minor_version: minor_version,
+            interface_id: interface_id,
+            type: type,
+            ownership: ownership,
+            aggregation: aggregation,
+            description: description,
+            doc: doc
+          } = Enum.into(interface_row, %{})
+
+          %InterfaceDocument{
+            name: name,
+            major_version: major_version,
+            minor_version: minor_version,
+            interface_id: interface_id,
+            type: InterfaceType.from_int(type),
+            ownership: Ownership.from_int(ownership),
+            aggregation: Aggregation.from_int(aggregation),
+            mappings: %{},
+            description: description,
+            doc: doc
+          }
+        end)
+
+      {:ok, mappings} = fetch_mappings(client)
```

Comment:
if fetch mappings fails and `{:error, <something>}` is returned, this will cause the pattern-match to fail and raise, which will be returned to the user as a "500 - internal server error".
Ideally we want to wrap this in a `with` statement so the error is returned to the calling function and will be processed accordingly

something like:

```elixir
  def get_interfaces_details(client) do
    with {:ok, interfaces} <- fetch_interfaces_witout_mappings(client),
          {:ok, mappings} <- fetch_mappings(client) do

        # merge interfaces and mappings
    end
  end

  defp fetch_interfaces_witout_mappings(client) do
    # fetch interfaces
  end

  defp fetch_mappings(client) do
    # fetch mappings
  end
```